sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsException.cs

```diff
@@ -216,7 +217,13 @@ public enum FailureReason
             ServiceTimeout,
 
             /// <summary>There was a general communications error encountered when interacting with the Azure Event Hubs service.</summary>
-            ServiceCommunicationProblem
+            ServiceCommunicationProblem,
+
+            /// <summary>A client was forcefully disconnected from an Event Hub instance.  This typically occurs when another consumer with higher <see cref="PartitionPublishingOptions.OwnerLevel" /> asserts ownership over the partition and producer group.</summary>
+            ProducerDisconnected,
```

Comment:
There is an open item for discussion in the design around this member.  Including it for now.